src/components/HTMLEngineProvider/BaseHTMLEngineProvider.tsx

```diff
@@ -1,88 +1,90 @@
-import PropTypes from 'prop-types';
 import React, {useMemo} from 'react';
-import {defaultHTMLElementModels, RenderHTMLConfigProvider, TRenderEngineProvider} from 'react-native-render-html';
-import _ from 'underscore';
+import {TextProps} from 'react-native';
+import {HTMLContentModel, HTMLElementModel, RenderHTMLConfigProvider, TRenderEngineProvider} from 'react-native-render-html';
 import useThemeStyles from '@hooks/useThemeStyles';
 import convertToLTR from '@libs/convertToLTR';
 import singleFontFamily from '@styles/utils/fontFamily/singleFontFamily';
 import * as HTMLEngineUtils from './htmlEngineUtils';
 import htmlRenderers from './HTMLRenderers';
+import {HTMLEngineProviderProps} from './types';
 
-const propTypes = {
+type BaseHTMLEngineProviderProps = HTMLEngineProviderProps & {
     /** Whether text elements should be selectable */
-    textSelectable: PropTypes.bool,
+    textSelectable?: boolean;
 
     /** Handle line breaks according to the HTML standard (default on web)  */
-    enableExperimentalBRCollapsing: PropTypes.bool,
-
-    children: PropTypes.node,
-};
-
-const defaultProps = {
-    textSelectable: false,
-    children: null,
-    enableExperimentalBRCollapsing: false,
+    enableExperimentalBRCollapsing?: boolean;
 };
 
 // We are using the explicit composite architecture for performance gains.
 // Configuration for RenderHTML is handled in a top-level component providing
 // context to RenderHTMLSource components. See https://git.io/JRcZb
 // Beware that each prop should be referentialy stable between renders to avoid
 // costly invalidations and commits.
-function BaseHTMLEngineProvider(props) {
+function BaseHTMLEngineProvider({textSelectable = false, children, enableExperimentalBRCollapsing = false}: BaseHTMLEngineProviderProps) {
     const styles = useThemeStyles();
 
     // Declare nonstandard tags and their content model here
+    /* eslint-disable @typescript-eslint/naming-convention */
     const customHTMLElementModels = useMemo(
         () => ({
-            edited: defaultHTMLElementModels.span.extend({
+            edited: HTMLElementModel.fromCustomModel({
```

Comment:
@ruben-rebelo Please:
1. Explain why these updates are necessary
2. Test it extensively to make sure it works the same way as before.

Reply:
1. This was changed because the extend function do not allow the change of the tagName, in order to change the tagName of the component we must generate a new element based on the contentModel.
2. I'll test it a bit more since next-steps were wrongly created, but so far all the elements seem to be correctly changed on the database and displayed on the platforms. Will let you know when I'm sure nothing will break.

Reply:
@VickyStash Regarding the testing, I can confirm that the elements are displayed correctly and store correctly on Onyx database.